scripts/reporting/ncanda_quality_control_script.py

```diff
@@ -0,0 +1,315 @@
+			if type(row.get('visit_date')) != str:
+				error = dict(subject_site_id = idx[0],
+					event_name = idx[1],
+					error = 'ERROR: Visit date missing.'
+					)
+	return error
+
+def WAIS_score_verification(idx,row):
+	"""
+	Verifies whether the wais_rawscore was computed correctly.
+	"""
+	# visit_ignore____yes with value 0 is not ignored
+	error = dict()
+	if math.isnan(row.get('exclude')):
+		if row.get('visit_ignore___yes') != 1:
+			# form is not missing if form_missing if value nan or zero
+			if row.get('np_wais4_missing') != 1:
+				if row.get('np_wais4_rawscore_computed') == row.get('np_wais4_rawscore_diff(correct)'):
+					if row.get('np_wais4_rawscore_diff(correct)') != 0:
+						error = dict(subject_site_id = idx[0],
+							visit_date = row.get('visit_date'),
+							event_name = idx[1],
+							error = 'ERROR: WAIS score is not verified'
+							)
+	return error
+
+def youth_report_sex(idx,row, field_missing, field_sex):
+	"""
+	Checks whether or not sex was entered correctly in the Youth Report
+	"""
+	# visit_ignore____yes with value 0 is not ignored
+	error = dict()
+	if math.isnan(row.get('exclude')):
+		if row.get('visit_ignore___yes') != 1:
+			# np is not missing if field_missing if value nan or zero
+			if row.get(field_missing) != 1:
+				if row.get('sex') != row.get(field_sex):
+					error = dict(subject_site_id = idx[0],
+								visit_date = row.get('visit_date'),
+								event_name = idx[1],
+								field = field_sex,
+								error = 'ERROR: SEX and SEX in YOUTHREPORT do not match.'
+								)
+	return error
+
+
+def main(args=None):
+    project_entry = get_project_entry()
+    project_df = data_entry_fields(fields,project_entry,'1y_visit_arm_1')
+    error = []
+
+    for idx, row in project_df.iterrows():
+        for f in form_fields:
+            check = missing_form(idx,row,f[0],f[1])
+            if check:
+                error.append(check)
+        for np in np_gpeg_fields:
+            check = np_groove_check(idx,row,'np_gpeg_missing',np[0],np[1])
+            if check:
+                error.append(check)
+        check = fourteen_days_MRI_report(idx,row)
+        if check:
+            error.append(check)
+        check = cnp_dob(idx, row)
+        if check:
+            error.append(check)
+        check = missing_mri_stroop(idx, row)
+        if check:
+            error.append(check)
+        for s in saliva_fields:
+            check = missing_saliva_sample(idx,row,s[0],s[1])
+            if check:
+                error.append(check)
+        check = visit_data_missing(idx,row)
+        if check:
+            error.append(check)
+        check = WAIS_score_verification(idx,row)
+        if check:
+            error.append(check)
+        for f in fields_sex:
+			check = youth_report_sex(idx,row,f[0],f[1])
+			if check:
+				error.append(check)
+
+
+    for e in error:
+        if e != 'null':
+            print json.dumps(e, sort_keys = True)
+
+
+if __name__ == '__main__':
+    import argparse
```

Comment:
as an argument to argparse, you can check other scripts for examples... arm may not make sense here though since the checks are pretty arm-specific so you can just add visit and case

parser.add_arguement('-a', '--arm', choices=['standard', 'sleep..', ... ], default='standard', help...)